tests/unittest_observer/FaceliftObserverTest.cpp

```diff
@@ -0,0 +1,76 @@
+#include <gtest/gtest.h>
+#include "IPCProxyBase.h"
+#include "InterfaceBase.h"
+#include <QSignalSpy>
+
+namespace {
+
+using namespace facelift;
+
+class Counter
+{
+public:
+    Counter() { m_value = 0; }
+
+    void incValue() {
+        m_value++;
+    }
+    int getValue() const {
+       return m_value;
+    }
+private:
+    int m_value;
+};
+
+class IPCProxyBaseTest : public ::testing::Test
+{
+public:
+    Counter c1;
+    Counter c2;
+    StandartObserver  < std::function<void()> > * obs1 = new StandartObserver  < std::function<void()> > (std::bind(&Counter::incValue, &c1) );
+    SingleTimeObserver< std::function<void()> > * obs2 = new SingleTimeObserver< std::function<void()> > (std::bind(&Counter::incValue, &c2) );
+
+    IPCProxyBase<InterfaceBase> proxyBase{nullptr};
+
+    ~IPCProxyBaseTest() {
+        delete obs1;
+        obs1 = nullptr;
+        delete obs2;
+        obs2 = nullptr;
+    }
+};
+
+TEST_F(IPCProxyBaseTest, testObservers)
+{
+    // Set observers to proxy
+    const auto expected = QVector<IObserver*>{obs1, obs2,};
+    proxyBase.m_readyObserver.setObservers(expected);
+    const auto actual = proxyBase.m_readyObserver.getObservers();
+
+    // Check installed observers
+    EXPECT_EQ(expected.size(), actual.size());
+    EXPECT_EQ(expected[0], actual[0]);
+    EXPECT_EQ(expected[1], actual[1]);
+
+    // Check values before calling a signal
+    ASSERT_EQ(c1.getValue(), 0); // for StandartObserver
+    ASSERT_EQ(c2.getValue(), 0); // for SingleTimeObserver
+
+    // Check handle of signal
+    QSignalSpy spy(&proxyBase.m_readyObserver, &IsReadyObserver::readyChanged );
+    ASSERT_EQ( spy.isValid(), true);
+    spy.clear();
+
+    // Generate signal
+    ASSERT_EQ(spy.count(), 0);
+    proxyBase.readyChanged();
+    proxyBase.readyChanged();
+    proxyBase.readyChanged();
+    ASSERT_EQ(spy.count(), 3);
+
+    // Check values after signal call
+    ASSERT_EQ(c1.getValue(), 3); // for StandartObserver
+    ASSERT_EQ(c2.getValue(), 1); // for SingleTimeObserver
+}
```

Comment:
You do not test the functionality. Also I am not quite sure how to use your observer, which was meant to be similar to discussed "callOnceReady" and "callOnReady".

Reply:
I added comments for the test, which briefly describes the sequence of work and interaction between "Proxy", "IsReadyObserver" and observers for "callOnReady", "callOnceReady". Instead "Proxy" there may be a different type and signal.

Reply:
The main goal to have a flexible service present as independent/extern component, that react on certain state of one object of inspection and execute vary functors either once or permanent.
In general for a system objects a certain kind of types, that has the "ready" state, it is necessary to provide the ability to track this state using the "IsReadyObserver" component and an arbitrary number of observers of two types: "callOnReady" and "callOnceReady", which are functors. The ready state is determined by a specific signal that emitted from  inspected object. The "IsReadyObserver" is the watcher of the signal of interest in "Proxy" and contain observers with functors to execute.
CAUTION: No one signal must be lost or overlooked.

Main steps:
- Creation of main objects: object for inspection "Proxy" (this is one of more types), object for observation "IsReadyObserver", objects for executing "Observers".
- Installing objects/function to be executed in observers on depence of require "callOnReady" or  "callOnceReady".
- Setting up to "IsReadyObserver" to watch out for signal in "Proxy".
- Catching signal and execution of functors.
- Completion of work.

Testing:
1) Functionality check and main life cycle of components (creation, execution, completion)
2) Processing of erroneous or incorrect data
3) Memory leaks and undefined behavior
4) Load testing for performance (performance, latency, metrics).
5) Stress testing for failure

Possible points 1-4 are minimal necessary.